### PR TITLE
FIX: Improve get_all_insertion_loss_list 

### LINKED ISSUE #7338: Bug located in analysis_hf.py
### Before submitting the issue

- [x] I have searched among the existing issues
- [x] I am using a Python virtual environment

### Description of the bug

When get_all_insertion_loss_list is run with no inputs, FEXT and NEXT are returned alongside insertion loss, when only insertion loss is expected. 

### Steps To Reproduce

- Call h3d.get_all_insertion_loss_list()

### Which Operating System are you using?

Windows

### Which Python version are you using?

3.12

### Installed packages

aiohappyeyeballs==2.6.1
aiohttp==3.12.9
aiosignal==1.3.2
annotated-types==0.7.0
ansys-api-edb==0.2.0
ansys-edb-core==0.2.0
ansys-pythonnet==3.1.0rc6
ansys-tools-visualization-interface==0.9.2
anyio==4.9.0
argon2-cffi==25.1.0
argon2-cffi-bindings==21.2.0
arrow==1.3.0
asgiref==3.9.1
asttokens==3.0.0
async-lru==2.0.5
attrs==25.3.0
babel==2.17.0
beautifulsoup4==4.13.4
bidict==0.23.1
bleach==6.2.0
blosc2==3.3.4
certifi==2025.4.26
cffi==1.17.1
charset-normalizer==3.4.2
cloudpickle==3.1.1
clr_loader==0.2.7.post0
colorama==0.4.6
comm==0.2.2
contourpy==1.3.2
cycler==0.12.1
debugpy==1.8.14
decorator==5.2.1
defusedxml==0.7.1
Django==5.2.4
emi_utilities @ file:///D:/Repositories/EMI_automation-main/dist/emi_utilities-0.2.dev1.tar.gz#sha256=a77dd6d7a1752be6e8d782786433437d930c5547c7301a1a2ff2fffd1ce121cc
et_xmlfile==2.0.0
executing==2.2.0
ezdxf==1.4.3
fast_simplification==0.1.11
fastjsonschema==2.21.1
fonttools==4.58.1
fpdf2==2.8.3
fqdn==1.5.1
frozenlist==1.6.2
geopandas==1.1.0
grpcio==1.70.0
h11==0.16.0
httpcore==1.0.9
httpx==0.28.1
idna==3.10
imageio==2.37.0
iniconfig==2.1.0
ipycanvas==0.13.3
ipyevents==2.0.2
ipykernel==6.29.5
ipython==8.37.0
ipython-genutils==0.2.0
ipyvtklink==0.2.3
ipywidgets==7.8.5
isoduration==20.11.0
jedi==0.19.2
Jinja2==3.1.6
json5==0.12.0
jsonpointer==3.0.0
jsonschema==4.24.0
jsonschema-specifications==2025.4.1
jupyter-events==0.12.0
jupyter-lsp==2.2.5
jupyter_client==8.6.3
jupyter_core==5.8.1
jupyter_server==2.16.0
jupyter_server_terminals==0.5.3
jupyterlab==4.4.3
jupyterlab_pygments==0.3.0
jupyterlab_server==2.27.3
jupyterlab_widgets==1.1.11
kiwisolver==1.4.8
markdown-it-py==3.0.0
MarkupSafe==3.0.2
matplotlib==3.10.3
matplotlib-inline==0.1.7
mdurl==0.1.2
meshio==5.3.5
mistune==3.1.3
more-itertools==10.7.0
msgpack==1.1.0
multidict==6.4.4
nbclient==0.10.2
nbconvert==7.16.6
nbformat==5.10.4
ndindex==1.10.0
nest-asyncio==1.6.0
networkx==3.5
notebook==7.4.3
notebook_shim==0.2.4
numexpr==2.10.2
numpy==2.2.6
openpyxl==3.1.5
osmnx==2.0.3
overrides==7.7.0
packaging==25.0
pandas==2.2.3
pandocfilters==1.5.1
parso==0.8.4
pillow==11.2.1
platformdirs==4.3.8
pluggy==1.6.0
plumbum==1.9.0
pooch==1.8.2
prometheus_client==0.22.1
prompt_toolkit==3.0.51
propcache==0.3.1
protobuf==4.25.8
psutil==7.0.0
pure_eval==0.2.3
py-cpuinfo==9.0.0
pyaedt @ file:///C:/Users/mpapac/Downloads/pyaedt-main.zip#sha256=f47606a1297276ed9033c56143dfcb961c0a356a61cc8d53bf5bc96a2d993658
pycparser==2.22
pydantic==2.10.6
pydantic_core==2.27.2
pyedb==0.69.0
Pygments==2.19.1
pyogrio==0.11.0
pyparsing==3.2.3
pyproj==3.7.1
pytest==8.4.0
python-dateutil==2.9.0.post0
python-engineio==4.12.3
python-json-logger==3.3.0
python-socketio==5.14.2
pytz==2025.2
pyvista==0.44.2
pywin32==310
pywinpty==2.0.15
PyYAML==6.0.2
pyzmq==26.4.0
referencing==0.36.2
requests==2.32.3
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rich==14.0.0
rpds-py==0.25.1
rpyc==6.0.2
rtree==1.4.0
scikit-rf==1.7.0
scipy==1.15.3
scooby==0.10.1
Send2Trash==1.8.3
setuptools==80.9.0
shapely==2.1.1
simple-websocket==1.1.0
six==1.17.0
sniffio==1.3.1
soupsieve==2.7
spyder-kernels==3.1.1
sqlparse==0.5.3
stack-data==0.6.3
tables==3.10.2
terminado==0.18.1
tinycss2==1.4.0
toml==0.10.2
tomli_w==1.2.0
tornado==6.5.1
traitlets==5.14.3
trame==3.10.2
trame-client==3.9.1
trame-common==1.0.0
trame-server==3.4.2
trame-vtk==2.8.17
trame-vuetify==3.0.1
types-python-dateutil==2.9.0.20250516
typing_extensions==4.14.0
tzdata==2025.2
uri-template==1.3.0
urllib3==2.4.0
vtk==9.3.1
wcwidth==0.2.13
webcolors==24.11.1
webencodings==0.5.1
websocket-client==1.8.0
websockets==15.0.1
widgetsnbextension==3.6.10
wslink==2.3.4
wsproto==1.2.0
xmltodict==1.0.4
yarl==1.20.0

### PR DESCRIPTION
**Pull request by Samuelopez-ansys**
## Description
This pull request improves the `get_all_insertion_loss_list` method in `analysis_hf.py` to provide more predictable and flexible behavior when generating insertion loss lists, and adds comprehensive test coverage for different usage scenarios. The changes clarify how driver and receiver lists are paired, handle edge cases more gracefully, and make the API easier to understand with examples.

Enhancements to insertion loss calculation logic:

* Refactored the pairing logic in `get_all_insertion_loss_list` so that when drivers and receivers are different lists, insertion losses are paired by index, and when they are the same, all unique combinations are generated. This makes the output more predictable and aligns with typical use cases.

* Improved error handling: When the lengths of the driver and receiver lists do not match, the method now returns an empty list instead of `False`, providing a more consistent and Pythonic API.

Documentation improvements:

* Added detailed usage examples to the docstring of `get_all_insertion_loss_list`, illustrating typical scenarios and expected outputs.

## Issue linked
Close #7338 

## Checklist
- [ ] I have tested my changes locally.
- [ ] I have added necessary documentation or updated existing documentation.
- [ ] I have followed the coding style guidelines of this project.
- [ ] I have added appropriate tests (unit, integration, system).
- [ ] I have reviewed my changes before submitting this pull request.
- [ ] I have linked the issue or issues that are solved by the PR if any.
- [ ] I have agreed with the Contributor License Agreement ([CLA](https://developer.ansys.com/form/cla-acceptance)).

### CODE REVIEW

File: src/ansys/aedt/core/application/analysis_hf.py [hunk 1]
```diff
@@ -181,18 +195,24 @@ def get_all_insertion_loss_list(
...
-                        spar.append(f"S({i},{j})")
+            # When drivers and receivers are different lists, pair by index
+            if drivers != receivers:
+                for idx, driver in enumerate(drivers):
+                    receiver = receivers[idx]
+                    if driver != receiver:
+                        if math_formula:
+                            spar.append(f"{math_formula}(S({driver},{receiver}))")
+                        else:
+                            spar.append(f"S({driver},{receiver})")
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/application/analysis_hf.py]: Since you know that they are of the same length, I think you could just use something like
```python
for driver, receiver in zip(drivers, receivers):
    ...
```

File: src/ansys/aedt/core/application/analysis_hf.py [hunk 2]
```diff
@@ -181,18 +195,24 @@ def get_all_insertion_loss_list(
...
+            # When drivers and receivers are the same list, generate all insertion loss combinations
+            else:
+                for i, driver in enumerate(drivers):
```
Comment (SMoraisAnsys) [inline src/ansys/aedt/core/application/analysis_hf.py]: You can problably work with zip here too but that might be more difficult to read so no issue with sticking with the current approach unless you want to give a try to zip everywhere ;)

**Comment by Samuelopez-ansys** [comment]
@SMoraisAnsys You convinced me, in addition, I added "combinations" which I read is very efficient, even if it is not a heavy task, but I like it, it is more pythonic.
